llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -976,8 +976,11 @@ void VPWidenIntrinsicRecipe::execute(VPTransformState &State) {
     CI->getOperandBundlesAsDefs(OpBundles);
 
   CallInst *V = State.Builder.CreateCall(VectorF, Args, OpBundles);
-
-  setFlags(V);
+  // vector-predication intrinsics only accept FMF flags, while vector intrinsic
+  // can support all flags.
+  bool VPIntrinsic = VPIntrinsic::isVPIntrinsic(VectorIntrinsicID);
+  if ((VPIntrinsic && isa<FPMathOperator>(V)) || !VPIntrinsic)
```

Comment:
`if (!VPIntrinsic || isa<FPMathOperator>(V))` is better

Reply:
I'm thinking whether we need this check. 
Is it possible to use the existing constructor to pass in an `OpType` other than `FPMathOp` when constructing vp intrinsic?